aws/resource_aws_acm_certificate_validation.go

```diff
@@ -83,11 +84,38 @@ func resourceAwsAcmCertificateValidationCreate(d *schema.ResourceData, meta inte
 	})
 }
 
-func resourceAwsAcmCertificateCheckValidationRecords(validation_record_fqdns []interface{}, cert *acm.CertificateDetail) error {
+func resourceAwsAcmCertificateCheckValidationRecords(validation_record_fqdns []interface{}, cert *acm.CertificateDetail, conn *acm.ACM) error {
 	expected_fqdns := make([]string, len(cert.DomainValidationOptions))
+
+	if len(cert.DomainValidationOptions) == 0 {
+		input := &acm.DescribeCertificateInput{
+			CertificateArn: cert.CertificateArn,
+		}
+		err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+			log.Printf("[DEBUG] Certificate domain validation options empty, retrying")
```

Comment:
Nitpick: Having the cert ARN in the log message would make it IMO more helpful.

Reply:
Oh good catch! Will add